src/components/composer/composer/composer.ts

```diff
@@ -172,9 +173,16 @@ export class Composer extends Component<ComposerProps, SpreadsheetChildEnv> {
         // render left
         assistantStyle += `right:0px;`;
       }
-      return (assistantStyle += `width:${ASSISTANT_WIDTH}px;`);
+      return assistantStyle;
+    }
+
+    const { width } = this.env.model.getters.getSheetViewDimensionWithHeaders();
+    if (width < ASSISTANT_WIDTH + this.composerRef.el!.getBoundingClientRect().left) {
+      assistantStyle += `right: 0px;`;
+      return assistantStyle;
```

Comment:
I won't keep this return here, as maybe one days we will add more style after this if, and for the moment it won't change anything as we directly returns after the if :)